### ISSUE TITLE
A `policy` kwarg for metadata

### ISSUE THREAD

**Issue by FirefoxMetzger** [imageio/imageio#760]
Sometimes writers produce corrupt metadata when writing images. At the moment ImageIO doesn't have a clear policy for handling this, so some (most) of the time things will break whereas other times things may be silently ignored or a warning may be raised. How should we handle this consistently?

Obviously things shouldn't break, but the ideal response is not so clear. During discussion in https://github.com/imageio/imageio/pull/753 we identified 3 candidates:

- `raise`/`strict`: Halt when reader encounters invalid metadata and raise an exception
- `warn`: Warn when reader encounters invalid metadata and ignore the field.
- `ignore`: Silently ignore any invalid metadata

Our thinking was to add a `policy` kwarg to `metadata` (at least for tifffile), but I'm not sure which of these options is the better default (debating between `raise` and `warn`). As such, I'm putting this up for discussion here. If you have an opinion or ideas how to handle this, feel free to make them known :)

(As usual, if nobody voices any objections, I will simply go with my own ideas 😉 )

**Comment by hmaarrfk** [comment]
I would like to push for "warn" during "loading" since often the user doesn't control their data or the loading path. imageio may be used within an other library that doesn't expose all the keyword arguments.

During save, it is easier to reject the user's input, and they are in more control of their data. So it may be appropriate to raise an error there.

**Comment by almarklein** [comment]
I agree with @hmaarrfk. Graceful degradation seems a better default than being strict. Bonus points for making the policy settable :)
